### ISSUE TITLE
Understanding the radio buttons in Persistent Highlights

### ISSUE THREAD

**Issue by claremacrae** [nothingislost/obsidian-dynamic-highlights#31]
Apologies if I missed it in the docs, but can you please say that the radio buttons mean in "Persistent Highlights" against regex rules:

- matches
- parent line
- start
- end

Perhaps tool tips/hover tips might help?

**Comment by nothingislost** [comment]
Agreed that this needs better documentation. For now, those toggles control how and where CSS classes are applied.

- matches: any string matching the search query will be wrapped in a `<span class="[highlightName]">`
- parent: any line containing a string that matches the search query will marked with the highlighter name `<div class="cm-line [highlightName]">`
- start: any string matching the search query will be prefixed with an empty span `<span class="[highlightName]-start"></span>`
- end: any string matching the search query will be suffixed with an empty span `<span class="[highlightName]-end"></span>`
- capture groups: only appears when using regex queries and will wrap any named capture groups with a class matching the capture group name

**Comment by claremacrae** [comment]
Thank you! I wouldn't have guessed that, but I can see how it's really useful.

**Comment by stefanopagliari** [comment]
@nothingislost, do you have any example of hour I can use the "parent" filter?
I would like to put a border around tasks.
I have been identifying tasks using `^- \[.\] .*$` but the border is applied to individual elements, not the entire line. If I switch from "matches" to "parent", the regex is not being picked up. I guess I am misunderstanding the functioning.
<img width="232" alt="image" src="https://user-images.githubusercontent.com/30759878/175785129-938a1dcc-f1e5-437b-9127-58b50b1ab1ff.png">
